src/helm/proxy/clients/auto_client.py

```diff
@@ -104,14 +104,18 @@ def provide_api_key():
                 )
                 client = create_object(client_spec)
             elif organization == "neurips":
-                client = HTTPModelClient(tokenizer=tokenizer, cache_config=cache_config)
+                client = HTTPModelClient(cache_config=cache_config)
             elif organization == "openai":
                 from helm.proxy.clients.openai_client import OpenAIClient
 
                 org_id = self.credentials.get("openaiOrgId", None)
                 api_key = self.credentials.get("openaiApiKey", None)
+                # HACK: OpenAIClient's tokenizer is only used by GPT-3.5 and GPT-4, which uses the
+                # cl100k_base tokenizer. So we always set the tokenizer to cl100k_base even if
+                # the given OpenAI model should actually use p50k_base or r50k_base.
+                # TODO: Get the tokenizer name from ModelDeployment instead.
                 client = OpenAIClient(
-                    tokenizer=tokenizer,
+                    tokenizer=self._get_tokenizer("anthropic/claude"),
```

Comment:
Fix this to cl100k

Reply:
Fixed; thanks for catching.

Reply:
Tokenizer is now provided by dependency injection after #1903